packages/dds/tree/src/test/feature-libraries/modular-schema/modularChangeFamily.spec.ts

```diff
@@ -1049,6 +1068,291 @@ describe("ModularChangeFamily", () => {
+		});
+
+		it("with existing builds", () => {
+			const changeA: HasRemovedRootsRefs = {
+				shallow: [a1],
+				nested: [],
+			};
+			const input: ModularChangeset = {
+				fieldChanges: new Map([
+					[brand("fA"), { fieldKind: removedRootsFieldKind, change: brand(changeA) }],
+				]),
+				builds: new Map([
+					[undefined, new Map([[brand(2), node2Chunk]])],
+					[aMajor, new Map([[brand(2), node2Chunk]])],
+				]),
+			};
+
+			const expected: ModularChangeset = {
+				fieldChanges: new Map([
+					[brand("fA"), { fieldKind: removedRootsFieldKind, change: brand(changeA) }],
+				]),
+				builds: new Map([
+					[undefined, new Map([[brand(2), node2Chunk]])],
+					[
+						aMajor,
+						new Map([
+							[brand(1), node1Chunk],
+							[brand(2), node2Chunk],
+						]),
+					],
+				]),
+			};
+
+			const withBuilds = addMissingBuilds(
+				makeAnonChange(input),
+				getDetachedNode,
+				mockFieldKinds,
+			);
+			assert.deepEqual(withBuilds, expected);
+		});
+	});
+
+	describe("filters builds", () => {
+		const aMajor = mintRevisionTag();
+		const bMajor = mintRevisionTag();
+		const b1 = { major: bMajor, minor: 1 };
+
+		const node2 = singleJsonCursor(2);
+		const node2Chunk = treeChunkFromCursor(node2);
+		const node3 = singleJsonCursor(3);
+		const node3Chunk = treeChunkFromCursor(node3);
+
+		it("with a valid relevant removed root", () => {
+			const changeB: HasRemovedRootsRefs = {
+				shallow: [b1],
+				nested: [],
+			};
+			const input: ModularChangeset = {
+				fieldChanges: new Map([
+					[
+						brand("fB"),
+						{
+							fieldKind: removedRootsFieldKind,
+							change: brand(changeB),
+							revision: bMajor,
+						},
+					],
+				]),
+				builds: new Map([
+					[
+						aMajor,
+						new Map([
+							[brand(1), node1Chunk],
+							[brand(2), node2Chunk],
+						]),
+					],
+					[bMajor, new Map([[brand(1), node3Chunk]])],
+				]),
+			};
+
+			const expected: ModularChangeset = {
+				fieldChanges: new Map([
+					[
+						brand("fB"),
+						{
+							fieldKind: removedRootsFieldKind,
+							change: brand(changeB),
+							revision: bMajor,
+						},
+					],
+				]),
+				builds: new Map([[bMajor, new Map([[brand(1), node3Chunk]])]]),
+			};
+
+			const filtered = filterBuilds(makeAnonChange(input), mockFieldKinds);
+			assert.deepEqual(filtered, expected);
+		});
+
+		it("with no relevant removed roots in the changeset", () => {
+			const input: ModularChangeset = {
+				fieldChanges: new Map([]),
+				builds: new Map([
+					[
+						aMajor,
+						new Map([
+							[brand(1), node1Chunk],
+							[brand(2), node2Chunk],
+						]),
+					],
+					[bMajor, new Map([[brand(1), node3Chunk]])],
+				]),
+			};
+
+			const expected: ModularChangeset = {
+				fieldChanges: new Map([]),
+			};
+
+			const filtered = filterBuilds(makeAnonChange(input), mockFieldKinds);
+			assert.deepEqual(filtered, expected);
+		});
+
+		it("only when a removed root is not relevant", () => {
```

Comment:
"only when a removed root is not relevant" is not really a claim that this test enforces since doesn't check that it does remove non-relevant roots.
```suggestion
		it("unless it refers to a relevant root", () => {
```